### ISSUE TITLE
Composer environment creation fails with 403 due to missing SA rights assigments

### ISSUE THREAD

**Issue by philip-harvey** [terraform-google-modules/terraform-google-composer#57]
### TL;DR

Composer environment creation fails when using the module due it multiple SA rights assignments that are not included in the module.  

### Expected behavior

Module grants the rights required, or at least documents the rights required if it doesn't grant them.

Specifically:
Composer service account requires "roles/composer.worker" and "roles/iam.serviceAccountUser"

For Shared VPC configuration:
Google APIs service account in the project  (_projectnumber_@cloudservices.gserviceaccount.com) requires  on the shared VPC project:
"roles/compute.networkUser"

GKE service account in the project (service-_projectnumber_@container-engine-robot.iam.gserviceaccount.com) requires  on the shared VPC project:
"roles/compute.networkUser" and "roles/container.hostServiceAgentUser"

Composer agent service account in the project (service-_projectnumber_@cloudcomposer-accounts.iam.gserviceaccount.com) requires on the shared VPC project: 
"roles/composer.sharedVpcAgent"

### Observed behavior

Creation of environment fails with 403

### Terraform Configuration

```hcl
resource "google_service_account" "composer" {
  account_id   = "composer-sa"
  display_name = "Service Account for Composer"
  project_id =   var.project_id
}

module "composer_environment" {
  source                               = "terraform-google-modules/composer/google//modules/create_environment_v2"
  version                              = "~> 3.2"
  project_id                           =   var.project_id
  composer_env_name                    = "composer"
  region                               = var.region
  image_version                        = "composer-2.0.20-airflow-2.2.5"
  composer_service_account             = google_service_account.composer.name
  network_project_id                   = var.shared_vpc_project
  subnetwork_region                    = var.region
  use_private_environment              = true
  enable_private_endpoint              = true
  network                              = data.google_compute_network.shared_vpc.name
  subnetwork                           = data.google_compute_subnetwork.shared_vpc_primary_subnet.name
  cloud_composer_connection_subnetwork = data.google_compute_subnetwork.shared_vpc_primary_subnet.self_link
  master_ipv4_cidr                     = "10.1.2.3/28"
  service_ip_allocation_range_name     = "composer-svc"
  pod_ip_allocation_range_name         = "composer-pod"
  environment_size                     = "ENVIRONMENT_SIZE_SMALL"
}
```

### Terraform Version

```sh
Terraform v1.2.4
+ provider registry.terraform.io/hashicorp/google v4.28.0
+ provider registry.terraform.io/hashicorp/google-beta v4.28.0
```
